Update integrate_with_Kobiton docs and CLI command

xuankimtran opened the pull request:
I updated Integrate with Kobiton docs and the CLI command for Kobiton device name. @nhpthuy @AntoineTouch

nhpthuy commented on pages/katalon-studio/docs/console-mode-execution.md:
```diff
@@ -494,6 +505,11 @@ katalonc -noSplash -runMode=console -projectPath="C:\Users\Katalon Studio\Projec
       </tr>
       <tr>
+      <td>-kobitonDeviceName=[yourKobitionDeviceName]</td>
+			<td>Dynamically change Kobiton device (Available in 8.1.0 and later)</td>
```
It should be "From version 8.1.0,..."

nhpthuy commented on pages/katalon-studio/docs/console-mode-execution.md:
```diff
@@ -172,6 +172,17 @@ Here's the list of options supported for the `katalonc` commands for Katalon Stu
+      <td>-maxFailedTests=&lt;T&gt;</td>
+			<td>
+				<ul>
+					<li><span data-preserver-spaces="true">Available from v8.1.0, you can set a condition to stop test suite/ test suite collection execution based on the number of test failures.</span></li>
```
Should be "From version 8.1.0,..."

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -36,29 +36,30 @@ First you need to install the [Kobiton Integration](https://store.katalon.com/pr
 
-   * **Kobiton Server**: The Kobiton server to be integrated with Katalon Studio.
+   * **Kobiton Server**: The Kobiton server to be integrated with Katalon Studio. 
+   > In **8.1.0 onwards**, you can customize the remote server protocol or pathname. Katalon Studio use HTTP protocol by default. If any prevention sending the basic authentication credential through HTTP occurs, you can input a remote URL with HTTPS instead.
```
.. remote server protocol and pathname. Katalon Studio uses... If your environment prevents sending basic authentication credentials through HTTP, you can input a remote URL with HTTPS instead.

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -36,29 +36,30 @@ First you need to install the [Kobiton Integration](https://store.katalon.com/pr
+> Kobiton integration is enabled, and you have adjusted your existing test scripts accordingly. Refer to this section: [Enable Kobiton Integration](https://docs.katalon.com/katalon-studio/docs/enable-kobiton-integration.html) for more details.
 
-There will be cases you want to use additional desired capabilities for Kobiton devices, such as using the `appWaitActivity` capability to troubleshoot an issue related to starting an application (check it out [here](/display/KD/Troubleshooting+automated+mobile+testing)). The tips below can help you to overcome this issue.
+This section provides tips on how to overcome the issue when you want to use additional desired capabilities for Kobiton devices, such as using the `appWaitActivity` capability to troubleshoot an issue related to starting an application. See [Troubleshooting automated mobile testing](https://docs.katalon.com/katalon-studio/docs/troubleshooting-automated-mobile-testing.html).
```
You can add

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -36,29 +36,30 @@ First you need to install the [Kobiton Integration](https://store.katalon.com/pr
+This section provides tips on how to overcome the issue when you want to use additional desired capabilities for Kobiton devices, such as using the `appWaitActivity` capability to troubleshoot an issue related to starting an application. See [Troubleshooting automated mobile testing](https://docs.katalon.com/katalon-studio/docs/troubleshooting-automated-mobile-testing.html).
 
-1. [Grab desired capabilities](https://docs.kobiton.com/automation-testing/automation-testing-with-kobiton/) generated from the Kobiton portal of the device you want to use and paste it to your test script.  
+1. [Grab desired capabilities](https://docs.kobiton.com/automation-testing/automation-testing-with-kobiton/) generates from the Kobiton portal of the device you want to use and paste it to your test script.  
```
Please fix this step by step with clearer steps as discussed

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -156,18 +157,26 @@ There will be cases you want to use additional desired capabilities for Kobiton
+
+> From **8.1.0 onwards**, you can change Kobiton's **Custom Device Name** to select the correct device on Katalon Studio. If your Kobiton devices don't have any custom name, Katalon Studio uses the devices' original name when running tests. See Kobiton documentation on [Device Custom Name](https://support.kobiton.com/hc/en-us/articles/360055588512-Device-Custom-Name-)
+
+After you set Device Custom Name in Kobiton Portal, you can check your Kobiton devices list in Katalon Studio by open **Mobile Recorder/Spy > Select Kobiton Device**.
```
Once you have set your Device Custom Names in Kobiton Portal, you can check the Kobiton devices list in Katalon Studio. Go to Katalon Studio, then open ....

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -210,15 +219,17 @@ Mobile.closeApplication()
-7\. On the Kobiton Favorite Devices screen, a list of **Favorite Devices** from _Step 4_ will be displayed in _Device Name_.  
+<img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/tutorials/integrate_with_kobiton/Run.png" alt="run with Kobiton Device" width=70%>
+
+7\. On the Kobiton Favorite Devices screen, a list of **Favorite Devices** from Step 4 displays in **Device Name**.  
```
I'd remove the mention to Step 4. Not necessary

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -210,15 +219,17 @@ Mobile.closeApplication()
 Select your preferred device and click **Ok**.
 
-You can also modify this list by updating your **Favorite Devices** from [https://portal.kobiton.com/devices](https://portal.kobiton.com/devices).
+You can also modify this list by updating your **Favorite Devices** from [Kobiton Portal](https://portal.kobiton.com/devices).
```
Notes box

xuankimtran commented:
I've changed "Condition to stop" to "Terminate execution additionally" and adjusted the content following Antoine's feedback last week. Please check it once more time @AntoineTouch @nhpthuy

AntoineTouch commented on pages/katalon-studio/docs/console-mode-execution.md:
```diff
@@ -583,31 +597,33 @@ We recommend using the Command Builder to generate commands quickly and precisel
-      <img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/docs/console-mode-execution/update-web-driver-automatically.png" width=70%>
+      <img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/docs/console-mode-execution/condition%20to%20stop%20-%202.png" alt="execution configurations" width=70%>
+
+      Available from **version 8.1.0 onwards**, you can terminate execution additionally with the option: **Terminate the execution once the total number of test failures reaches the input threshold**.
```
Remove available for correct grammar - you can check proper use of time locators.

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -21,45 +21,45 @@ redirect_from:
 
-* In **7.8 onwards**, enter you username and API Key, and click **Test Connection**.
-   <img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/docs/enable-kobiton-integration/sso-kobiton.png">
+* In **7.8 onwards**, enter you username and API Key, then click **Test Connection**.
```
From... onwards:

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -21,45 +21,45 @@ redirect_from:
 
-1. [Grab desired capabilities](https://docs.kobiton.com/automation-testing/automation-testing-with-kobiton/) generated from the Kobiton portal of the device you want to use and paste it to your test script.  
-    ![](https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/docs/desired-capabilities-for-kobiton-devices/Screen-Shot-2018-07-05-at-11.40.52.png)  
+1. Grab [desired capabilities](https://docs.kobiton.com/automation-testing/automation-testing-with-kobiton/): In the Kobiton portal, select the device you want to use. Click **Automation setting** and copy the desired dapabilities.
```
check spelling of capabilities. Rephrase "grab".

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -156,18 +156,27 @@ There will be cases you want to use additional desired capabilities for Kobiton
 ## Mobile Testing with Kobiton Devices
 
-You need to install the Kobiton Integration plugin and enable the integration. If you haven't configured the integration yet, please refer to [this document](https://docs.katalon.com/katalon-studio/docs/enable-kobiton-integration.html) for instructions.
+You need to install and enable the Kobiton Integration plugin. If you haven't configured the integration yet, refer to this section: [Enable Kobiton Integration](https://docs.katalon.com/katalon-studio/docs/enable-kobiton-integration.html) for instructions.
```
I would remove "need". Say: Install... or Make sure you have ... enabled.

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -156,18 +156,27 @@ There will be cases you want to use additional desired capabilities for Kobiton
-You need to install the Kobiton Integration plugin and enable the integration. If you haven't configured the integration yet, please refer to [this document](https://docs.katalon.com/katalon-studio/docs/enable-kobiton-integration.html) for instructions.
+You need to install and enable the Kobiton Integration plugin. If you haven't configured the integration yet, refer to this section: [Enable Kobiton Integration](https://docs.katalon.com/katalon-studio/docs/enable-kobiton-integration.html) for instructions.
+
+Please follow the instructions below to execute your Katalon Studio automation test with Kobiton mobile devices.
```
Remove please

AntoineTouch replied:
You can say instead: To do this, follow these instructions:

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -156,18 +156,27 @@ There will be cases you want to use additional desired capabilities for Kobiton
+
+1\. Navigate to [Kobiton Portal](https://portal.kobiton.com) and log in with your Kobiton username and password.
+
+2\. In Kobiton, upload your app to Kobiton App Repository. See Kobiton documentation on [Managing Application](https://support.kobiton.com/hc/en-us/articles/360056028911-Managing-Applications-).
```
to the Kobiton App Repository.

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -156,18 +156,27 @@ There will be cases you want to use additional desired capabilities for Kobiton
+<img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/tutorials/integrate_with_kobiton/Automation-snippet.png" alt="Automation snippet" width=70%>
 
-2\. In Kobiton, upload your app to [Kobiton App Repository](https://docs.kobiton.com/#managing-apps).![Kobiton App Repository](https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/tutorials/integrate_with_kobiton/Kobiton-App-Repository.png)
+4\. Click on the **Devices** menu. Select your device, then hover to the top left corner of the tab to **Mark as favorite**.
```
You still need to click after hovering if I remember correctly

AntoineTouch replied:
You can say: click on the _star_ icon in the top right corner to mark as favourite

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -210,15 +219,20 @@ Mobile.closeApplication()
 Select your preferred device and click **Ok**.
 
-You can also modify this list by updating your **Favorite Devices** from [https://portal.kobiton.com/devices](https://portal.kobiton.com/devices).
+>**Note**:
```
Notes, even in singular. This is a fixed format

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -21,45 +21,45 @@ redirect_from:
-1. [Grab desired capabilities](https://docs.kobiton.com/automation-testing/automation-testing-with-kobiton/) generated from the Kobiton portal of the device you want to use and paste it to your test script.  
-    ![](https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/docs/desired-capabilities-for-kobiton-devices/Screen-Shot-2018-07-05-at-11.40.52.png)  
+1. Copy the [desired capabilities](https://docs.kobiton.com/automation-testing/automation-testing-with-kobiton/): In the Kobiton portal, select the device you want to use. Click **Automation setting** and copy the desired capabilities.
+Open Katalon Studio, paste these desired capabilities to your test script. 
```
then paste

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -156,18 +156,27 @@ There will be cases you want to use additional desired capabilities for Kobiton
+
+1\. Navigate to [Kobiton Portal](https://portal.kobiton.com) and log in with your Kobiton username and password.
+
+2\. In Kobiton, upload your app to the Kobiton App Repository. See Kobiton documentation on [Managing Application](https://support.kobiton.com/hc/en-us/articles/360056028911-Managing-Applications-).
```
Applications

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -156,18 +156,27 @@ There will be cases you want to use additional desired capabilities for Kobiton
+<img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/tutorials/integrate_with_kobiton/Automation-snippet.png" alt="Automation snippet" width=70%>
 
-2\. In Kobiton, upload your app to [Kobiton App Repository](https://docs.kobiton.com/#managing-apps).![Kobiton App Repository](https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/tutorials/integrate_with_kobiton/Kobiton-App-Repository.png)
+4\. Click on the **Devices** menu. Select your device, then hover and click on the star icon in the top right corner to **Mark as favorite**.
```
delete hover

nhpthuy commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -21,45 +21,45 @@ redirect_from:
 
-* In **7.8 onwards**, enter you username and API Key, and click **Test Connection**.
-   <img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/docs/enable-kobiton-integration/sso-kobiton.png">
+* From **7.8 onwards**: enter you username and API Key, then click **Test Connection**.
```
pls use the format x.x.x for versioning, in this case: 7.8.0

nhpthuy commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -21,45 +21,45 @@ redirect_from:
-
-   * **Kobiton Server**: The Kobiton server to be integrated with Katalon Studio.
+   * **Kobiton Server**: The Kobiton server to be integrated with Katalon Studio. 
+   > In **8.1.0 onwards**, you can customize the remote server protocol and pathname. Katalon Studio uses HTTP protocol by default. If your environment prevents sending basic authentication credentials through HTTP, you can input a remote URL with HTTPS instead.
```
From version **8.1.0 onwards**

nhpthuy commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -21,45 +21,45 @@ redirect_from:
+   > In **8.1.0 onwards**, you can customize the remote server protocol and pathname. Katalon Studio uses HTTP protocol by default. If your environment prevents sending basic authentication credentials through HTTP, you can input a remote URL with HTTPS instead.
    * **API Key**: The token to be used by Katalon Studio when exchanging API messages with Kobiton server. You can generate more keys in [Kobiton API Settings](https://portal.kobiton.com/settings/keys).
 
 * **Before 7.8**, enter your Kobiton account in the **Authentication** form and click **Connect**. Katalon Studio retrieves the information for Kobiton integration automatically.
```
Prior to version 7.8.0

nhpthuy commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -21,45 +21,45 @@ redirect_from:
+    <img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/docs/enable-kobiton-integration/image2017-6-29-163A543A3.png" alt="enable kobiton integration before 7.8" width=70%>
   
-3. Click **Apply** and then **OK** when you are done with the settings.
+4. When you are done with the settings, click **Apply and close**.
```
consider using "configuration" instead of "settings"

nhpthuy commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -21,45 +21,45 @@ redirect_from:
 > **Prerequisites**
 >
-> Kobiton integration is enabled, and you have adjusted your existing test scripts accordingly. Refer to this [guide](/display/KD/Mobile+Testing+with+Kobiton+Devices) for more details.
+> Kobiton integration is enabled, and you have adjusted your existing test scripts accordingly. Refer to this section: [Enable Kobiton Integration](https://docs.katalon.com/katalon-studio/docs/enable-kobiton-integration.html) for more details.
```
Refer to [Enable Kobiton Integration](https://docs.katalon.com/katalon-studio/docs/enable-kobiton-integration.html) for more details

AntoineTouch commented on pages/katalon-studio/docs/console-mode-execution.md:
```diff
@@ -583,31 +597,33 @@ We recommend using the Command Builder to generate commands quickly and precisel
-      <img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/docs/console-mode-execution/update-web-driver-automatically.png" width=70%>
+      <img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/docs/console-mode-execution/condition%20to%20stop%20-%202.png" alt="execution configurations" width=70%>
+
+      From **version 8.1.0 onwards**, you can terminate execution conditionally with the option: **Terminate the execution once the total number of test failures reaches the input threshold**.
```
Replace the complicated words terminate execution conditionally with simpler wording.

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -156,18 +156,27 @@ There will be cases you want to use additional desired capabilities for Kobiton
+<img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/tutorials/integrate_with_kobiton/add-app.png" width="100%">
 
-Please follow the instructions to execute your Katalon Studio automation test with Kobiton mobile devices.
+3\. From the Repository view, click on the drop-down icon on the right corner. Copy the app ID (for example, **2229394** as shown below) and save it to use later in Step 5.
```
don't mention other step numbers as it just tempts people to break away from the flow. My immediate thought when I see this is  "What step am I at, and how far is step 5?", which means I lose track.

You can just say "copy the app ID to your clipboard for later use."

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -156,18 +156,27 @@ There will be cases you want to use additional desired capabilities for Kobiton
+<img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/tutorials/integrate_with_kobiton/app-ID.png" width=70%>
 
-2\. In Kobiton, upload your app to [Kobiton App Repository](https://docs.kobiton.com/#managing-apps).![Kobiton App Repository](https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/tutorials/integrate_with_kobiton/Kobiton-App-Repository.png)
+4\. Click on the **Devices** menu. Select your device, then click on the flag icon in the top right corner to **Mark as favorite**.
```
_flag_ 

As per Microsoft style guide, don't say "icon" if you can help it.

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -156,18 +156,27 @@ There will be cases you want to use additional desired capabilities for Kobiton
+<img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/tutorials/integrate_with_kobiton/mark-as-favourite.png" width=70%>
 
-4\. Click on the **Devices** menu and select your devices by "_Mark as favorite"._![Kobiton Devices menu](https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/tutorials/integrate_with_kobiton/Devices.png)
+> From **8.1.0 onwards**, you can change Kobiton's **Custom Device Name** to select the correct device on Katalon Studio. If your Kobiton devices don't have any custom name, Katalon Studio uses the devices' original name when running tests. See Kobiton documentation on [Device Custom Name](https://support.kobiton.com/hc/en-us/articles/360055588512-Device-Custom-Name-)
```
version 8.1.0 onwards

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -210,15 +219,20 @@ Mobile.closeApplication()
+
+<img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/tutorials/integrate_with_kobiton/Favorite-Devices.png" alt="Kobiton Favorite Devices" width=70%>
 
 8\. Once Katalon Studio is finished, automation test execution will be uploaded to Kobiton. Navigate to **Sessions** menu to view:
```
9. To view X, Navigate to Y in Kobiton.

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -210,15 +219,22 @@ Mobile.closeApplication()
+
+<img src="https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/tutorials/integrate_with_kobiton/Favorite-Devices.png" alt="Kobiton Favorite Devices" width=70%>
+
+8\. Once Katalon Studio is finished, automation test execution will be uploaded to Kobiton. 
```
Katalon Studio is finished with what?

Remove automation. The phrase is automation testing (for what we do) or automatic/automated when you are talking about specific tests. but this is redundant here.

Is this supposed to be test execution _results_? Not sure what is uploaded or what is run where.

AntoineTouch commented on pages/katalon-studio/tutorials/integrate_with_kobiton.md:
```diff
@@ -210,15 +219,22 @@ Mobile.closeApplication()
+8\. Once Katalon Studio is finished, automation test execution will be uploaded to Kobiton. 
 
-![Kobiton Favorite Devices](https://github.com/katalon-studio/docs-images/raw/master/katalon-studio/tutorials/integrate_with_kobiton/Favorite-Devices.png)
+9\. To view the session, navigate to **Sessions** in Kobiton.
```
I would not make this as step 9, but just as a sentence afterwards: "you can view the session in the **Sessions** section in Kobiton.

AntoineTouch replied:
The reason for this is that the title for this section is : 
## Mobile Testing with Kobiton Devices

And so your last step in the step-by-step should be about testing, not viewing.

You could otherwise change the header to something like "Perform and view Mobile Tests using Kobiton Devices"

Padoplin reviewed (approved):
Pls resolve my comment, then it's OK to be merged.

Padoplin commented on pages/katalon-studio/docs/console-mode-execution.md:
```diff
@@ -172,6 +172,17 @@ Here's the list of options supported for the `katalonc` commands for Katalon Stu
+      <td>-maxFailedTests=&lt;T&gt;</td>
+			<td>
+				<ul>
+					<li><span data-preserver-spaces="true">From version 8.1.0, you can terminate a test suite/ test suite collection execution based on the number of test failures. (Only for Katalon Studio Enterprise users)</span></li>
```
This function is available for Katalon Studio user too. So pls remove _(Only for Katalon Studio Enterprise users)_.

Padoplin replied:
Version info is not consistently formatted. Sometimes it's bold, sometimes it's normal text.

Padoplin reviewed (approved):
Resolve my comments, then it's ready.
